src/WinGetServer/WinGetServerManualActivation_Client.cpp

```diff
@@ -0,0 +1,166 @@
+#include <filesystem>
+#include <shlobj_core.h>
+
+#if USE_PROD_WINGET_SERVER
+const std::wstring_view s_ServerExePath = L"Microsoft\\WindowsApps\\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\\WindowsPackageManagerServer.exe";
+#else
+const std::wstring_view s_ServerExePath = L"Microsoft\\WindowsApps\\WinGetDevCLI_8wekyb3d8bbwe\\WindowsPackageManagerServerDev.exe";
+#endif
+
+_Must_inspect_result_
+_Ret_maybenull_ _Post_writable_byte_size_(size)
+void* __RPC_USER MIDL_user_allocate(_In_ size_t size)
+{
+    return malloc(size);
+}
+
+void __RPC_USER MIDL_user_free(_Pre_maybenull_ _Post_invalid_ void* ptr)
+{
+    if (ptr)
+    {
+        free(ptr);
+    }
+}
+
+std::filesystem::path GetKnownFolderPath(const KNOWNFOLDERID& id)
+{
+    wil::unique_cotaskmem_string knownFolder = nullptr;
+    THROW_IF_FAILED(SHGetKnownFolderPath(id, KF_FLAG_NO_ALIAS | KF_FLAG_DONT_VERIFY | KF_FLAG_NO_PACKAGE_REDIRECTION, NULL, &knownFolder));
+    return knownFolder.get();
+}
+
+struct FreeWithRpcStringFree { void operator()(RPC_CSTR* in) { RpcStringFreeA(in); } };
+using UniqueRpcString = std::unique_ptr<RPC_CSTR, FreeWithRpcStringFree>;
+
+struct DeleteWithMidlFree { void operator()(void* m) { MIDL_user_free(m); } };
+using UniqueMidl = std::unique_ptr<BYTE, DeleteWithMidlFree>;
+
+void InitializeRpcBinding()
+{
+    std::string protocol = "ncacn_np";
+    std::string endpoint = "\\pipe\\WinGetServerManualActivation_" + GetUserSID();
+
+    unsigned char* binding = nullptr;
+    UniqueRpcString bindingPtr;
+
+    RPC_STATUS status = RpcStringBindingComposeA(nullptr, GetUCharString(protocol), nullptr, GetUCharString(endpoint), nullptr, &binding);
+    THROW_HR_IF(HRESULT_FROM_WIN32(status), status != RPC_S_OK);
+    bindingPtr.reset(&binding);
+
+    status = RpcBindingFromStringBindingA(binding, &WinGetServerManualActivation_IfHandle);
+    THROW_HR_IF(HRESULT_FROM_WIN32(status), status != RPC_S_OK);
+}
+
+HRESULT LaunchWinGetServerWithManualActivation()
+{
+    const std::filesystem::path& localAppDataPath = GetKnownFolderPath(FOLDERID_LocalAppData);
+    const std::filesystem::path& serverExePath = localAppDataPath / s_ServerExePath;
+    std::wstring commandLineInput = std::wstring{ serverExePath } + L" --manualActivation";
+
+    STARTUPINFO info = { sizeof(info) };
+    PROCESS_INFORMATION processInfo;
+    if (!CreateProcessW(NULL, &commandLineInput[0], NULL, NULL, FALSE, 0, NULL, NULL, &info, &processInfo))
+    {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
+
+    // Wait for manual reset event from server before proceeding with COM activation.
+    HANDLE eventHandle = CreateEventW(NULL, TRUE, FALSE, L"WinGetServerStartEvent");
+    RETURN_HR_IF_NULL(PEER_E_EVENT_HANDLE_NOT_FOUND, eventHandle);
+
+    DWORD waitResult = WaitForSingleObject(eventHandle, INFINITE);
+    if (waitResult != 0)
+    {
+        return ERROR_SERVICE_NEVER_STARTED;
+    }
+
+    CloseHandle(processInfo.hProcess);
+    CloseHandle(processInfo.hThread);

```

Comment:
Should we close thread first then process?

Reply:
Changed to using wil::unique_process_information to take care of handle cleanups.